modules/web/js/ballerina/ast/resource-definition.js

```diff
@@ -269,7 +270,7 @@ class ResourceDefinition extends ASTNode {
             if (firstWorkerIndex > -1 && _.isNil(index)) {
                 index = firstWorkerIndex;
             }
-            Object.getPrototypeOf(this.constructor.prototype).addChild.call(this, child, index);
+            Object.getPrototypeOf(this.constructor.prototype).addChild.call(this, child, index, ignoreTreeModifiedEvent);
```

Comment:
seems like this particular line is common to both if and the else clause. If so we can get it out from the clauses